src/maker.rs

```diff
@@ -391,4 +417,44 @@ mod tests {
 
         assert_eq!(result, TakeRequestDecision::RateSucks);
     }
+
+    #[test]
+    fn rate_updated_if_rate_update_with_same_value() {
+        let mut maker = Maker {
+            mid_market_rate: MidMarketRate {
+                value: Rate::try_from(1.0).unwrap(),
+                ..Default::default()
```

Comment:
:wrench: It would be better to instantiate the current mid market rate and the new mid market rate with specific values, so that you can be sure that the test is sound. Otherwise you're relying on now to produce two different values when called consecutively, which may vary between test executions.

Reply:
I think they are, this might be from an outdated version.